shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1106,16 +1122,49 @@ - (void)deleteBackward {
     [self replaceRange:_selectedTextRange withText:@""];
 }
 
+- (void)accessibilityElementDidBecomeFocused {
+  if ([self accessibilityElementIsFocused]) {
+    // For most of the cases, this flutter text input view should never
+    // receive the focus. If we do receive the focus, we make the best effort
+    // to send the focus back to the real text field.
+    __strong UIAccessibilityElement* textInput =
+        FlutterTextInputView.backingTextInputAccessibilityObject;
+    FML_DCHECK(textInput);
+    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, textInput);
+  }
+}
+
+- (BOOL)accessibilityElementsHidden {
+  return !_accessibilityEnabled;
+}
+
+@end
+
+/**
+ * Hides `FlutterTextInputView` from iOS accessibility system so it
+ * does not show up twice, once where it is in the `UIView` hierarchy,
+ * and a second time as part of the `SemanticsObject` hierarchy.
+ *
+ * This prevents the `FlutterTextInputView` from receiving the focus
+ * due to swipping gesture.
+ *
+ * There are other cases the `FlutterTextInputView` may receive
+ * focus. One example is during screen changes, the accessibility
+ * tree will undergo a dramatic structural update. The Voiceover may
+ * decide to focus the `FlutterTextInputView` that is not involved
+ * in the structural update instead. If that happens, the
+ * `FlutterTextInputView` will make a best effort to direct the
+ * focus back to the `SemanticsObject`.
```

Comment:
Couldn't this result in duplicate Voiceover announcements?  One for the FlutterTextInputView and one SemanticsObject?

Reply:
that is true, that is why i put the first two fail safe and hope it can catch all of the corner cases.

If this text view does receive the focus, duplicate announcement is still much better than focus just stuck at this textview.

I have tried a number of different hack such as changing the trait and label of this text view to prevent voiceover from pronouncing any word when the text view is focused, but none of them work. If you have a textview without any trait or label, it will pronounce some random phrase such as "blue sky" when it is focused.